### ISSUE TITLE
About cross compile tflite 

### ISSUE THREAD

**Issue by zorosmith** [Hozzu/TFLite-aarch64-linux-with-delegate#2]
Hello, I hope this message finds you well. 

I’m currently working on a project where I need to cross-compile TensorFlow Lite (TFLite) on an x86-based Ubuntu system and deploy it on a 64-bit ARM-based Ubuntu 18.04 system. After compiling the library using both CMake and Bazel, I’ve encountered issues integrating the compiled libraries into my C++ project.

Specifically, I have tried using the cross-compilation versions of TFLite that were provided by the authors, but I’ve run into compatibility issues related to the version of glibc on Ubuntu 18.04. It appears that the glibc version on Ubuntu 18.04 is too old, causing a version mismatch when trying to link the compiled TFLite libraries. The error like this，
`/home/gordon/demo_test/a133/tflite-216/lib/libtensorflowlite.so: undefined reference to `pow@GLIBC_2.29'
/home/gordon/demo_test/a133/tflite-216/lib/libtensorflowlite.so: undefined reference to `exp@GLIBC_2.29'
/home/gordon/demo_test/a133/tflite-216/lib/libtensorflowlite.so: undefined reference to `log@GLIBC_2.29'`

I would greatly appreciate it if you could provide some guidance on how TensorFlow Lite should be cross-compiled for a 64-bit ARM architecture. Specifically, I’m interested in knowing which method you use for compiling TFLite, and if possible, could you provide a tutorial on how to successfully perform the cross-compilation?

Thank you for your time and support

**Comment by Hozzu** [comment]
The bazel of tensorflow automatically downloads aarch64_linux_toolchain when building it.
If the glibc version of your system is older than that of the toolchain, you will confront that error.
You should download aarch64_linux_toolchain that matches your system and replace the bazel's toolchain with it.

You can download the toolchain at ARM homepage.
https://developer.arm.com/downloads/-/gnu-a
https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads

The bazel's toolchain is downloaded into ~/.cache/bazel/.....
The directory is different across systems.

You should also edit tensorflow/tools/toolchains/embedded/arm-linux/cc_config.bzl.tpl to match your toolchain version.

If you use CMake to build tensorflow, it is easier to use the custom toolchain.
Create and use CMAKE_TOOLCHAIN_FILE option.

**Comment by zorosmith** [comment]
Thank you for your response. I followed the suggestions you provided for cross-compiling TensorFlow Lite, but unfortunately, the issue still persists.
After considering alternative options, I decided to switch to using ONNX Runtime instead. ONNX Runtime provides official precompiled versions for 64-bit ARM architecture, which allowed me to avoid the complexity of cross-compiling.
I was able to directly convert the TensorFlow-trained model to the .onnx format, and it's now ready for deployment.
Thanks again for your help.
